packages/ra-ui-materialui/src/list/datagrid/useDatagridContext.ts

```diff
@@ -0,0 +1,14 @@
+import { useContext } from 'react';
+import DatagridContext, { DatagridContextValue } from './DatagridContext';
+
+export const useDatagridContext = (): DatagridContextValue => {
```

Comment:
What we did to keep backward compatibility until v4 is make sure all hooks of this kind also accept props. If props containing the expected values are provided, we return them instead of the context.

Reply:
You can look at the `useListContext` hook in the core package for inspiration

Reply:
Done!